ros2doctor/ros2doctor/api/platform.py

```diff
@@ -59,12 +48,12 @@ def category(self):
     def check(self):
         """Check system platform against ROS 2 Distro."""
         result = Result()
-        distros = _check_platform_helper()
-        if not distros:
-            result.add_error('ERROR: Missing rosdistro info. Unable to check platform.')
+        try:
+            distros = _check_platform_helper()
+        except (AttributeError, URLError, KeyError):
+            result.add_error('Error: Unable to fetch rosdistro information online.')
```

Comment:
@claireyywang I'm not sure if giving back _less_ information to the user is a good path forward. I don't know if this was agreed elsewhere though.

Reply:
Thanks for reviewing! In `package.py`, the same logic is used and merged. I simplified the logic here to keep it consistent.  The errors are most likely to occur when either user's machine has no internet connection, or there is something wrong with `rosdistro` package and it's not returning the right urls. The first case is covered by the network check, while the latter they can't do much about and should probably file a ticket. I think it is reasonable to remove the lengthy warnings and replace with a higher level one.

Reply:
Oops, sorry for the delay @claireyywang ! 

I see. I will say though that not giving enough information doesn't make things any easier. Regular ROS users will file a ticket, but will be otherwise unable to provide any insight as to what happened, which ultimately puts the burden on us. Power ROS users likely won't wait for a maintainer reply round-trip and will instead be forced to remove that (quietly) 
 catch-all clause to see the actual error. I understand your point about consistency, but I can't help wondering if the code that should be changed is the one in `package.py`, and not the other way around. 

Anyways, if folks things this is the way, I won't stand in the way.

Reply:
No worries. I see what you mean and it makes sense to expose more info to users. I don't feel strongly about how it should be, so I'll close this and make changes to `package.py` instead.

Reply:
No worries. I see what you mean and it makes sense to expose more info to users. I don't feel strongly about how it should be, so I'll close this and make changes to package.py instead.